common/config/rush/version-policies.json

```diff
@@ -0,0 +1,80 @@
+/**
+ * This is configuration file is used for advanced publishing configurations with Rush.
+ * For full documentation, please see https://rushjs.io
+ */
+[
+  /**
+   * A "version policy" is a custom package versioning strategy for use with "rush publish".
+   * The strategy affects a set of projects that are specified using the "versionPolicyName" field
+   * in rush.json.
+   */
+  {
+    /**
+     * (Required) Indicates the kind of version policy being defined ("lockStepVersion" or "individualVersion").
+     *
+     * The "lockStepVersion" mode specifies that the projects will use "lock-step versioning".  This
+     * strategy is appropriate for a set of packages that act as selectable components of a
+     * unified product.  The entire set of packages are always published together, and always share
+     * the same NPM version number.  When the pacakges depend on other packages in the set, the
+     * SemVer range is usually restricted to a single version.
+     */
+    "definitionName": "lockStepVersion",
+
+    /**
+     * (Required) The name that will be used for the "versionPolicyName" field in rush.json.
+     * This name is also used command-line parameters such as "--version-policy"
+     * and "--to-version-policy".
+     */
+    "policyName": "MyBigFramework",
+
+    /**
+     * (Required) The current version.  All packages belonging to the set should have this version
+     * in the current branch.  When bumping versions, Rush uses this to determine the next version.
+     * (The "version" field in package.json is NOT considered.)
```

Comment:
I always thought this was a strange design choice

Reply:
I disagree:  If you store your data in 10 project folders, and two of them are out of sync with the others for some reason, how do you determine the version?

Having a unique well-defined source of truth is pretty useful.  We've considered in the future having "rush install" generate the package.json files as a build output.

---
In reply to: [217226468](https://github.com/Microsoft/rush-example/pull/2#discussion_r217226468) [](ancestors = 217226468)